### DISCUSSION TITLE
TLS-SNI?

### DISCUSSION THREAD

**Discussion by riaanswartdivi** [dotnet/MQTTnet#1692] [Q&A]
Hi

I've had a look at the source code of MQTTnet, specifically the version we use (3.0.16). I see no mention of TLS-SNI in there. I haven't used it yet, but I suppose I must specifiy a specific server name to connect to, and that doesn't seem to be in the interface. 

Is TLS-SNI supported as well, so we can connect to public HiveMQ brokers? 

Thanks
Riaan Swart
Divigraph

**Comment by Nukta101** [comment]
@chkr1011 is it possible to achieve this with the 3.0.16 or the latest version of MQTTnet?

**Comment by ArslanAli101** [comment]
From the MQTTNet release v4.2.1.781 a new property "TargetHost" is exposed in the class MqttClientTlsOptions that could be used to set a custom SNI.

**Reply by RiaanSDivigraph** [reply]
Thank you very much!
